opentelemetry/instrumentation/docker-compose.yaml

```diff
@@ -0,0 +1,29 @@
+version: '3'
+
+services:
+  app:
+    build: ./app
+    environment:
+      - OTEL_EXPORTER_OTLP_ENDPOINT=http://otelcol:4318
+      - OTEL_SERVICE_NAME=otel-quickstart-go
+    volumes:
+      - logs:/var/log:rw
+    depends_on:
+      - "otelcol"
+  otelcol:
+    image: otel/opentelemetry-collector-contrib:0.85.0
+    volumes:
+      - ./otel-collector-config.yaml:/etc/otelcol-contrib/config.yaml:ro
+      - $GOOGLE_APPLICATION_CREDENTIALS:/tmp/keys/gcp-credentials.json:ro
+      - logs:/var/log:rw
+    environment:
+      - GOOGLE_APPLICATION_CREDENTIALS=/tmp/keys/gcp-credentials.json
+      - PROJECT_ID=$PROJECT_ID
```

Comment:
Is this required or optional with ADC fallback?

Reply:
When I do `gcloud auth application-default login`, it doesn't appear to set my project.  I'm not sure if that is something special with my setup, or if I need to change the login command...

Reply:
I think it depends if we ADC gets auth/projectID from the metadata server vs credentials file in your home dir.

If `PROJECT_ID` is required in Cloud Shell (target env for quickstarts), can we use [mandatory environment variables](https://docs.docker.com/compose/compose-file/12-interpolation/#:~:text=Similarly%2C%20the%20following%20syntax%20allows%20you%20to%20specify%20mandatory%20variables%3A)? It prints out a nice message if the variable is unset

Reply:
made both environment variables required